### PR TITLE
feat(new-mls-conversation): add protocol to new group (AR-1775)

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
Added new UI Component [WireDropDown] to select group conversation protocol, now we can select the group protocol.

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/keypackage/KeyPackageRepository.kt [hunk 1]
```diff
@@ -37,10 +37,9 @@ class KeyPackageDataSource(
...
             userIds.map { userId ->
                 wrapApiRequest {
-                    keyPackageApi.claimKeyPackages(idMapper.toApiModel(userId))
-                }.flatMap { keyPackageList ->
-                    // TODO: filtering out key packages from the self user client (will be removed soon when BE is updated).
-                    Either.Right(keyPackageList.keyPackages.filter { it.clientID != selfClientId.value })
+                    keyPackageApi.claimKeyPackages(idMapper.toApiModel(userId), selfClientId.value)
+                }.flatMap {
+                    Either.Right(it.keyPackages)
                 }
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/keypackage/KeyPackageRepository.kt:43]: Small syntax suggestion  👍🏼
```suggestion
                }.map { it.keyPackages }
```

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/keypackage/KeyPackageApiImpl.kt [hunk 2]
```diff
@@ -4,18 +4,20 @@ import com.wire.kalium.network.AuthenticatedNetworkClient
...
+import io.ktor.client.request.parameter
 import io.ktor.client.request.post
 import io.ktor.client.request.setBody
 
 class KeyPackageApiImpl internal constructor(private val authenticatedNetworkClient: AuthenticatedNetworkClient) : KeyPackageApi {
 
     private val httpClient get() = authenticatedNetworkClient.httpClient
 
-    override suspend fun claimKeyPackages(user: UserId): NetworkResponse<ClaimedKeyPackageList> =
+    override suspend fun claimKeyPackages(user: UserId, selfClientId: String?): NetworkResponse<ClaimedKeyPackageList> =
```
Comment (MohamadJaara) [inline network/src/commonMain/kotlin/com/wire/kalium/network/api/keypackage/KeyPackageApiImpl.kt:16]: Is there a reason for selfClientId to be nullable?

**Review by MohamadJaara — COMMENTED** [review]
lgtm 🚀, only a minor syntax suggestion and a question about selfClientId being nullable

File: network/src/commonMain/kotlin/com/wire/kalium/network/api/keypackage/KeyPackageApiImpl.kt (hunk 2 above)
Comment (mchenani) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/keypackage/KeyPackageApiImpl.kt:16]: selfClientId is a queryParam, and also it's only needed when we are claiming keypackages for current account.

Comment (MohamadJaara) [reply network/src/commonMain/kotlin/com/wire/kalium/network/api/keypackage/KeyPackageApiImpl.kt:16]: So maybe we can add something to make it clear that passing the selfClientId will call the endpoint with `skip_own` quarry parameter.

**Review by MohamadJaara — APPROVED** [review]
lgtm 🚀

**Pull request merged**